### PR TITLE
public cli and docs for intial advanced restore

### PR DESCRIPTION
**Pull request by ryanfkeepers**
creates public docs and un-hides the advanced restore option cli flags.

---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #3562

### CODE REVIEW

File: src/cli/restore/restore.go [hunk 1]
```diff
@@ -34,15 +34,41 @@ func AddCommands(cmd *cobra.Command) {
 
 const restoreCommand = "restore"
 
+//nolint:lll
+const restoreCommandExamples = `# Restore emails to their original location
+corso restore exchange \
+	--backup 1234abcd-12ab-cd34-56de-1234abcd \
+	--destination '/'
```
Comment (gmatev) [inline src/cli/restore/restore.go:41]: We should have the examples be more realistic. I don't think restore of the full service to the original location is such an example. 

Would recommend restore of a specific item or folder instead.

File: src/cli/restore/restore.go [hunk 2]
```diff
@@ -34,15 +34,41 @@ func AddCommands(cmd *cobra.Command) {
...
+
+# Restore OneDrive to a folder named "my_latest_restore"
+corso restore onedrive \
+	--backup 1234abcd-12ab-cd34-56de-1234abcd \
+	--mail-folder '*' \
+	--destination my_latest_restore
```
Comment (gmatev) [inline src/cli/restore/restore.go:47]: Are we allowing non-root restores? Should this be `/my_latest_restore` ?

File: src/cli/restore/restore.go [hunk 3]
```diff
@@ -34,15 +34,41 @@ func AddCommands(cmd *cobra.Command) {
 
 const restoreCommand = "restore"
 
+//nolint:lll
+const restoreCommandExamples = `# Restore emails to their original location
+corso restore exchange \
+	--backup 1234abcd-12ab-cd34-56de-1234abcd \
+	--destination '/'
+
+# Restore OneDrive to a folder named "my_latest_restore"
```
Comment (gmatev) [inline src/cli/restore/restore.go:43]:
```suggestion
# Restore all one drive folders OneDrive to a root folder named "my_latest_restore"
```

File: src/cli/restore/restore.go [hunk 4]
```diff
@@ -34,15 +34,41 @@ func AddCommands(cmd *cobra.Command) {
...
+corso restore onedrive \
+	--backup 1234abcd-12ab-cd34-56de-1234abcd \
+	--mail-folder '*' \
+	--destination my_latest_restore
+
+# Restore calendar events, restoring a copy if the event already exists.
+corso restore exchange \
+	--backup 1234abcd-12ab-cd34-56de-1234abcd \
+	--event-calendar '*' \
```
Comment (gmatev) [inline src/cli/restore/restore.go:52]: Can we make this be a specific calendar example?

**Comment by gmatev** [comment]
I would also recommend adding a paragraph to the quick start flow that can cover the in-place restore after the existing non-destructive restore option. Would make things a bit more discoverable.

**Comment by gmatev** [comment]
Another thought is also adding a topic under `Usage` that is focused on `Restore Options`. This can cover the topic in a bit more detail which I think is needed.

File: src/cli/restore/restore.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/cli/restore/restore.go:41]: done, added `--email-folder '/inbox'` as the selector.

File: src/cli/restore/restore.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/cli/restore/restore.go:47]: destination always stores to the top level.  `my_latest` and `/my_latest` are treated as equivalent values.

File: src/cli/restore/restore.go (hunk 4 above)
Comment (ryanfkeepers) [reply src/cli/restore/restore.go:52]: Certainly

File: src/cli/restore/restore.go (hunk 2 above)
Comment (gmatev) [reply src/cli/restore/restore.go:47]: I would use the more explicit value in the example since it is more consistent with the in-place option

**Comment by ryanfkeepers** [comment]
@gmatev ready for a second review.  I've added all the asks except for the addition to `Usage`, because I'm not quite sure where the best place to add that will be.

**Comment by vkamra** [comment]
> @gmatev ready for a second review. I've added all the asks except for the addition to `Usage`, because I'm not quite sure where the best place to add that will be.

@ryanfkeepers - can add it under a "Restore Options" page under Usage

image

**Review by vkamra — APPROVED** [review]
LGTM. Approving so we can merge and iterate.

File: website/docs/setup/advanced-restore.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+named `my-latest-restore`. If that folder doesn't already exist, Corso will
+automatically create it. If it does exist, the restore will use the existing
+folder and restore the items again. This allows you to restore to the same folder
+multiple times.
+
+You can restore items back to their original location by setting the destination
+to `/`: `--destination /`. This skips the creation of a top-level folder, and
+all restored items will appear back in their location at the time of backup.
+
+### Known Issues
```
Comment (gmatev) [inline website/docs/setup/advanced-restore.md:36]:
```suggestion
### Limitations
```
Known issues makes it confusing with the overall Known issues

File: website/docs/setup/advanced-restore.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+all of the restored calendar events will appear in the `Restored` calendar. However,
+if you restore events in-place (`--destination /`) then all events will return to
+their original calendars.
+
+* When restoring Exchange Calendar Events to a destination folder, Events that were
+safe in different calendars may collide with each other in the destination calendar.
+
+## Item Collision Handling
+
+When restoring data into an existing folder, the items from the backup may conflict
```
Comment (gmatev) [inline website/docs/setup/advanced-restore.md:52]:
```suggestion
When restoring data into an existing folder, the items restored may conflict
```

File: website/docs/setup/advanced-restore.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+`Corso_Restore_<current-date-time>`. If you're concerned about data integrity
+then this is always the safest option, 
+
+In the above example, the onedrive restore will appear in a top-level folder
+named `my-latest-restore`. If that folder doesn't already exist, Corso will
+automatically create it. If it does exist, the restore will use the existing
+folder and restore the items again. This allows you to restore to the same folder
+multiple times.
+
+You can restore items back to their original location by setting the destination
```
[the remark this answers is not in the document]
Comment (gmatev) [reply website/docs/setup/advanced-restore.md:32]: You have a version of what I'm suggesting around collisions below

File: website/docs/setup/advanced-restore.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+## Skip
+
+The default behavior. When a collision is identified, no restore is attempted.
+
+## Copy
+
+Item collisions create a copy of the item in the backup. The copy holds the backup
+version of the item, leaving the current version unchanged. If necessary, changes
+item properties (such as filenames) to avoid additional collisions.
```
Comment (gmatev) [inline website/docs/setup/advanced-restore.md:79]: Should include an example of the name modifiers when this happens.

File: website/docs/setup/advanced-restore.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+The comparison uses item metadata (names, subjects, titles, etc), not item content.
+If the current `reports.txt` has different contents than the backup `reports.txt`,
+it still collides.
+
+Collisions can be handled with three different configurations: `Skip`, `Copy`,
+and `Replace`.
+
+## Skip
+
+The default behavior. When a collision is identified, no restore is attempted.
```
Comment (gmatev) [inline website/docs/setup/advanced-restore.md:73]:
```suggestion
The default behavior. When a collision is identified, the item is skipped and no restore is attempted.
```

**Review by gmatev — CHANGES_REQUESTED** [review]
@ryanfkeepers added some suggestions

File: website/docs/setup/advanced-restore.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+folder and restore the items again. This allows you to restore to the same folder
+multiple times.
+
+You can restore items back to their original location by setting the destination
+to `/`: `--destination /`. This skips the creation of a top-level folder, and
+all restored items will appear back in their location at the time of backup.
+
+### Known Issues
+
+* Destination won't create N-depth folder structures. `--destination a/b/c`
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply website/docs/setup/advanced-restore.md:38]: We can make it error if the destination looks hierarchical.  That's not difficult.

[the remark this answers is not in the document]
Comment (gmatev) [reply website/docs/setup/advanced-restore.md:38]: That seems like a good idea to me

File: website/docs/setup/advanced-restore.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply website/docs/setup/advanced-restore.md:48]: Done.  @gmatev Thanks for the review, second draft is ready to read.

[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply website/docs/setup/advanced-restore.md:48]: Done.  @gmatev Thanks for the review, second draft is ready to read.

**Review by gmatev — APPROVED** [review]

File: website/docs/setup/advanced-restore.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply website/docs/setup/advanced-restore.md:54]: Will do in a follow-up PR.

**Pull request merged**